### PR TITLE
OCaml 4.02.2 fix, and matching camlp4 update

### PR DESCRIPTION
**Pull request by avsm**
See discussion in #41432

### CODE REVIEW

File: Library/Formula/camlp4.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Camlp4 < Formula
   desc "Tool to write extensible parsers in OCaml"
   homepage "https://github.com/ocaml/camlp4"
-  url "https://github.com/ocaml/camlp4/archive/4.02.1+2.tar.gz"
-  sha1 "ad43587c2bdc46d5d0f028857f5ce90bc9ce7ab5"
-  version "4.02.1+2"
+  url "https://github.com/ocaml/camlp4/archive/4.02+6.tar.gz"
+  sha1 "6dd7e591dfde94c44576dba9b847279ffb92c889"
```
Comment (DomT4) [inline Library/Formula/camlp4.rb]: Let's use `sha256` here, Thanks! Everything is being updated en-mass soon, so it's nice to gradually reduce the amount that has to be changed first.

**Comment by mikemcquaid** [comment]
If you add something to the `camlp4` test that checks the OCaml version it'll be run everything we bump it.

File: Library/Formula/objective-caml.rb [hunk 2]
```diff
@@ -1,9 +1,23 @@
+# OCaml does not preserve binary compatibility across compiler releases,
+# so when updating it you should ensure that all dependent packages are
+# also updated by incrementing their revisions.
```
Comment (mikemcquaid) [inline Library/Formula/objective-caml.rb]: Is there a way we can verify this in the `test do` blocks for those packages so they fail?

**Comment by avsm** [comment]
@mikemcquaid there's no hard check in camlp4 on the minor version. It actually works fine with either 4.02.1 or 4.02.2 since the AST format hasn't changed.  It just needs to be recompiled if the version of OCaml changes.

I think that in Homebrew terms this means that we have to manually bump the revision of camlp4 every time ocaml changes, right?  There's no other way to express an exact version dependency.

File: Library/Formula/objective-caml.rb (hunk 2 above)
Comment (avsm) [inline Library/Formula/objective-caml.rb]: The only one that we can do is a local OPAM bootstrap.  I'll take a look at this.

**Comment by mikemcquaid** [comment]
> I think that in Homebrew terms this means that we have to manually bump the revision of camlp4 every time ocaml changes, right? There's no other way to express an exact version dependency.

Yeh, correct. I guess I mean is there a way you could e.g. add to the `test do` block something (even if it's just Ruby) to verify what you're saying we need to do?

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

File: Library/Formula/objective-caml.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/objective-caml.rb]: That's be great to add.

**Pull request closed without merging**